### DISCUSSION TITLE
Expose file-based route name for reference

### DISCUSSION THREAD

**Discussion by weyert** [vercel/next.js#25482] [Ideas]
### Describe the feature you'd like to request

Currently when using a custom server and you use a catch-all route to let Next.js handle its requests, you don't have anyway to get the original file-named route e.g. `blogs/posts/[id]` while I think this is useful for observability purposes.

This would allow to add `http.route` to tracing spans when using Next.js and the original request path

### Describe the solution you'd like

I would propose to either add a function to retrieve the route details from an existing request, e.g. `getRouteFromRequest(req)`

### Describe alternatives you've considered

I haven't found a good alternative to solve this request

**Comment by weyert** [comment]
Just noticed that @timneutkens has moved my feature request to Discussions without an explanation. 

Does anyone know how to achieve this getting the file-based route bit?

**Reply by timneutkens** [reply]
> without an explanation

Please read the issue template:

> Feature requests will be converted to the GitHub Discussions "Ideas" section.

https://github.com/vercel/next.js/issues/new?assignees=&labels=template%3A+story&template=3.feature_request.yml

**Reply by weyert** [reply]
Okay, all clear. Do you know if there is a way to get it? 

As it’s really useful metrics and traces to have the route name too
